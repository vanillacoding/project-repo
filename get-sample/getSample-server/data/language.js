// 옵셔널.. 나중에 추가..
{
 "kind": "youtube#i18nLanguageListResponse",
 "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/mgP15jzKT_Wf5_5P9IScmDEf--8\"",
 "items": [
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/GMrwiM1f-4KHxMka40cB3lysLgY\"",
   "id": "af",
   "snippet": {
    "hl": "af",
    "name": "Afrikaans"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/wOlCLE4kfCyCca9_ssuNDceE0yk\"",
   "id": "az",
   "snippet": {
    "hl": "az",
    "name": "Azerbaijani"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/1mc5QDsxoAICO6jscO-tFPgIu90\"",
   "id": "id",
   "snippet": {
    "hl": "id",
    "name": "Indonesian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/wsT9bPntXoLehs2KLJYbqnGE0BU\"",
   "id": "ms",
   "snippet": {
    "hl": "ms",
    "name": "Malay"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/d4-apDn0ZDft0Yoco6Lj3wJYH-g\"",
   "id": "bs",
   "snippet": {
    "hl": "bs",
    "name": "Bosnian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/hVswTFoMTSh7qztQXgsDjxgTETU\"",
   "id": "ca",
   "snippet": {
    "hl": "ca",
    "name": "Catalan"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/3Y90h664Ob7mJ0QMNKsBNUq4CU0\"",
   "id": "cs",
   "snippet": {
    "hl": "cs",
    "name": "Czech"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/ebVwqGAAMKQ8RnZHJ1v1zjSlLYc\"",
   "id": "da",
   "snippet": {
    "hl": "da",
    "name": "Danish"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/PGhdoQnaK-BWQU4ym2yZcuiqlvo\"",
   "id": "de",
   "snippet": {
    "hl": "de",
    "name": "German"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/jJeiNic4lStyfYgyad8EXXSH2JA\"",
   "id": "et",
   "snippet": {
    "hl": "et",
    "name": "Estonian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/2vzKH692FQxYp_M1_rNk1yIFV8Y\"",
   "id": "en-GB",
   "snippet": {
    "hl": "en-GB",
    "name": "English (United Kingdom)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/so_Y993qk4-_su7kjbFEx7nnUp4\"",
   "id": "en",
   "snippet": {
    "hl": "en",
    "name": "English"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/XWxnUXsW_IJhD5ZOwHqOQoxQwYY\"",
   "id": "es",
   "snippet": {
    "hl": "es",
    "name": "Spanish (Spain)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/l-bosHudQSmdi5HROuUr_YdsSqE\"",
   "id": "es-419",
   "snippet": {
    "hl": "es-419",
    "name": "Spanish (Latin America)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/SrBrtHLAGm7kadsQ4c5DgEs9WYo\"",
   "id": "es-US",
   "snippet": {
    "hl": "es-US",
    "name": "Spanish (United States)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/kv8FTUm9fJESlhhsA0VHgaoW_9o\"",
   "id": "eu",
   "snippet": {
    "hl": "eu",
    "name": "Basque"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/NzJjZ2TVm87kGGnoltxAYXp8O0o\"",
   "id": "fil",
   "snippet": {
    "hl": "fil",
    "name": "Filipino"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/TNUHp9LsiTmgH8dlEf9JNhV4ohw\"",
   "id": "fr",
   "snippet": {
    "hl": "fr",
    "name": "French"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/YfJYO1tiIE29JfeN_UW0Tb5sJIY\"",
   "id": "fr-CA",
   "snippet": {
    "hl": "fr-CA",
    "name": "French (Canada)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/P5dUevqZ6BlWyrDIDjZv60t4m7k\"",
   "id": "gl",
   "snippet": {
    "hl": "gl",
    "name": "Galician"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/vh1P5hcnWe2N_5cWm7RvNVyBFZI\"",
   "id": "hr",
   "snippet": {
    "hl": "hr",
    "name": "Croatian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/1X1Ghj1zxxsI79xZqe3q1X5IH9M\"",
   "id": "zu",
   "snippet": {
    "hl": "zu",
    "name": "Zulu"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/SZDTKBO2oYN6xtKmjQA010KV2dk\"",
   "id": "is",
   "snippet": {
    "hl": "is",
    "name": "Icelandic"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/GkgF_P9qdb7fTnwkgbxcyn6m3Rs\"",
   "id": "it",
   "snippet": {
    "hl": "it",
    "name": "Italian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/uNRd_FYvmSYFHYzF9aPsvAQsogQ\"",
   "id": "sw",
   "snippet": {
    "hl": "sw",
    "name": "Swahili"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/uPWPSnBy6yqRcKDBL1Qi4LnFoLg\"",
   "id": "lv",
   "snippet": {
    "hl": "lv",
    "name": "Latvian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/DawyQ-FGzpJzc_8cNRqFbT07v_E\"",
   "id": "lt",
   "snippet": {
    "hl": "lt",
    "name": "Lithuanian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/3V9J28NoQBu29aIOlAVChWF_Kns\"",
   "id": "hu",
   "snippet": {
    "hl": "hu",
    "name": "Hungarian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/lJOztqYOgdQvvfmB0K_DdWuUeq8\"",
   "id": "nl",
   "snippet": {
    "hl": "nl",
    "name": "Dutch"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/t7VHxMiLEpiTvTnjo23cptlIWTQ\"",
   "id": "no",
   "snippet": {
    "hl": "no",
    "name": "Norwegian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/C9uxQ_iPORF4ZWcc3WOcX1t70j0\"",
   "id": "uz",
   "snippet": {
    "hl": "uz",
    "name": "Uzbek"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/Nx2oULoIl34LYNPdcfpsu1H6lRU\"",
   "id": "pl",
   "snippet": {
    "hl": "pl",
    "name": "Polish"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/YdBgwFNd8IC3dSomi4FySFwlKJw\"",
   "id": "pt-PT",
   "snippet": {
    "hl": "pt-PT",
    "name": "Portuguese (Portugal)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/qiKLbq_oWL4zljYGomyqgkH-6QA\"",
   "id": "pt",
   "snippet": {
    "hl": "pt",
    "name": "Portuguese (Brazil)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/Pn58jAcxF7qX5ATMsX-ggAwuxOs\"",
   "id": "ro",
   "snippet": {
    "hl": "ro",
    "name": "Romanian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/UplXAMXI0ufZC1c4ug1P9GLxSz4\"",
   "id": "sq",
   "snippet": {
    "hl": "sq",
    "name": "Albanian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/Z3EiXrpvTORUdZkpL0tSXuGhwyU\"",
   "id": "sk",
   "snippet": {
    "hl": "sk",
    "name": "Slovak"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/OR2fBqiMEI-SgPkcbpUujYdt180\"",
   "id": "sl",
   "snippet": {
    "hl": "sl",
    "name": "Slovenian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/K_iJM4hro7crlz-7aGI1Y_SwjNc\"",
   "id": "sr-Latn",
   "snippet": {
    "hl": "sr-Latn",
    "name": "Serbian (Latin)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/c-zvGcAzJLR2YtImY-wpVGgQq8Q\"",
   "id": "fi",
   "snippet": {
    "hl": "fi",
    "name": "Finnish"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/zRiHhvImrdhnQR1mY6g7ze2abEM\"",
   "id": "sv",
   "snippet": {
    "hl": "sv",
    "name": "Swedish"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/SCTfVJMz9xZI2E44-7UJP5OMrbE\"",
   "id": "vi",
   "snippet": {
    "hl": "vi",
    "name": "Vietnamese"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/QhD4Q6EupU3gDCUaTu_GfQNZ0oc\"",
   "id": "tr",
   "snippet": {
    "hl": "tr",
    "name": "Turkish"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/KpMXRgwxyWN2een_d8ZmUKN49N4\"",
   "id": "be",
   "snippet": {
    "hl": "be",
    "name": "Belarusian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/2h9ZelGleOUWIFGsIwOZPF0db7g\"",
   "id": "bg",
   "snippet": {
    "hl": "bg",
    "name": "Bulgarian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/L7XQWQKAdB4RKyxcdT7_zF3Evqg\"",
   "id": "ky",
   "snippet": {
    "hl": "ky",
    "name": "Kyrgyz"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/oGC_uk6pN2Vvlm-Xpj2sOVqKKjI\"",
   "id": "kk",
   "snippet": {
    "hl": "kk",
    "name": "Kazakh"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/wYm6Y2DP6q9wzyLhrodUSCK5r4E\"",
   "id": "mk",
   "snippet": {
    "hl": "mk",
    "name": "Macedonian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/Wjr61C4VmckZjh4xwUab8y4f0RU\"",
   "id": "mn",
   "snippet": {
    "hl": "mn",
    "name": "Mongolian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/t2zC4SXLYVo5G2kfXk9-bePFRpk\"",
   "id": "ru",
   "snippet": {
    "hl": "ru",
    "name": "Russian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/Ay3s66nL913XBNAEOd3CNd5PjtY\"",
   "id": "sr",
   "snippet": {
    "hl": "sr",
    "name": "Serbian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/0brBZoxnBEUP-Du3i48gXWkA0S8\"",
   "id": "uk",
   "snippet": {
    "hl": "uk",
    "name": "Ukrainian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/JfQhv6NxmYwz5mDVNBbo5Zm6U8I\"",
   "id": "el",
   "snippet": {
    "hl": "el",
    "name": "Greek"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/KylLrdp_i6vpR-ykX-_e3qbVclA\"",
   "id": "hy",
   "snippet": {
    "hl": "hy",
    "name": "Armenian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/md5OauXuB2desazaJGSf4JO70_I\"",
   "id": "iw",
   "snippet": {
    "hl": "iw",
    "name": "Hebrew"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/Uz-9K5eVJ0MM2iz-ZCswjK4yqIs\"",
   "id": "ur",
   "snippet": {
    "hl": "ur",
    "name": "Urdu"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/mws5phUh7Qxy_uf06QWokL42X9g\"",
   "id": "ar",
   "snippet": {
    "hl": "ar",
    "name": "Arabic"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/GZemVMnoMUYY3wosO0nXf-_rk2o\"",
   "id": "fa",
   "snippet": {
    "hl": "fa",
    "name": "Persian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/S3eqTLKBnwQX2WvwcHkMiltFRx8\"",
   "id": "ne",
   "snippet": {
    "hl": "ne",
    "name": "Nepali"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/CsWmhcPj0SiaL2VFA72Bp7ohDSE\"",
   "id": "mr",
   "snippet": {
    "hl": "mr",
    "name": "Marathi"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/JCyuVDfWcOgtaZwswnZyy1AzR5I\"",
   "id": "hi",
   "snippet": {
    "hl": "hi",
    "name": "Hindi"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/dhFhF7UIAmlZRrmvX_erFuBQCP4\"",
   "id": "bn",
   "snippet": {
    "hl": "bn",
    "name": "Bangla"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/qSMM8n3SiMBRIEwoCukCVsN_U1M\"",
   "id": "pa",
   "snippet": {
    "hl": "pa",
    "name": "Punjabi"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/CamKnOPN6X6DO5N_w2M62uu-BBo\"",
   "id": "gu",
   "snippet": {
    "hl": "gu",
    "name": "Gujarati"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/UpkJlfs19zVrDL4jxmko2TFfeiI\"",
   "id": "ta",
   "snippet": {
    "hl": "ta",
    "name": "Tamil"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/snSOiVsugZraJQmQooFEUgwrvGg\"",
   "id": "te",
   "snippet": {
    "hl": "te",
    "name": "Telugu"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/pTwsDdi2u1LezXypsaalDTmGsos\"",
   "id": "kn",
   "snippet": {
    "hl": "kn",
    "name": "Kannada"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/j9vpsCSOrQBCuqW95C1wNgF_OPY\"",
   "id": "ml",
   "snippet": {
    "hl": "ml",
    "name": "Malayalam"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/4q_QN7GcwgP5b_DbB3EEc5xAW8U\"",
   "id": "si",
   "snippet": {
    "hl": "si",
    "name": "Sinhala"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/_E_YSjVSJtbo4nl97gnQer9POqA\"",
   "id": "th",
   "snippet": {
    "hl": "th",
    "name": "Thai"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/2jrl7H3eP2Z63pm8-Ag-NrenlDY\"",
   "id": "lo",
   "snippet": {
    "hl": "lo",
    "name": "Lao"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/f-CmZu2q-Huf-NSgtSZ31BNLcss\"",
   "id": "my",
   "snippet": {
    "hl": "my",
    "name": "Myanmar (Burmese)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/TCXP3ae7Akd-IrzsHK5lMTeFO64\"",
   "id": "ka",
   "snippet": {
    "hl": "ka",
    "name": "Georgian"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/0TSfvf4f7k04LUHD7rSbdwlcbrI\"",
   "id": "am",
   "snippet": {
    "hl": "am",
    "name": "Amharic"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/RUsS-Iy62HUcG2uy3VN0qZhMQy0\"",
   "id": "km",
   "snippet": {
    "hl": "km",
    "name": "Khmer"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/wD2SRT6G7gbFH07ePlumHAynSRo\"",
   "id": "zh-CN",
   "snippet": {
    "hl": "zh-CN",
    "name": "Chinese"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/6fRre896XUzNQjc89q329PaFKjE\"",
   "id": "zh-TW",
   "snippet": {
    "hl": "zh-TW",
    "name": "Chinese (Taiwan)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/MbipoDEFiRRUlYr5UzjpCwXXRMc\"",
   "id": "zh-HK",
   "snippet": {
    "hl": "zh-HK",
    "name": "Chinese (Hong Kong)"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/XBmnabKsnR1WSWcZvNxC2NvrLYo\"",
   "id": "ja",
   "snippet": {
    "hl": "ja",
    "name": "Japanese"
   }
  },
  {
   "kind": "youtube#i18nLanguage",
   "etag": "\"j6xRRd8dTPVVptg711_CSPADRfg/l06bH0pLscIVm87oyBnJ3aZR4Ts\"",
   "id": "ko",
   "snippet": {
    "hl": "ko",
    "name": "Korean"
   }
  }
 ]
}
