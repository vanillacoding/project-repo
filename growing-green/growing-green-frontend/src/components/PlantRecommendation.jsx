import React from 'react';
import styled from 'styled-components';

export default function PlantRecommendation({ plantsNames }) {
  return (
    <Container>
      <Title>사용자들이 가장 많이 키우는 식물 TOP 5</Title>
      {plantsNames?.map((name) => (
        <p key={name}>{name}</p>
      ))}
    </Container>
  );
}

const Container = styled.div`
  position: absolute;
  bottom: 2rem;
  display: flex;
  flex-direction: column;
  width: 600px;
  height: 350px;
  background: white;
  z-index: -1;
`;

const Title = styled.h1`
  margin: 2rem;
  font-size: 1.3em;
`;
