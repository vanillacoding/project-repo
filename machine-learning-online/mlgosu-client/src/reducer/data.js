import { combineReducers } from 'redux';
import { LOAD_DATA } from '../constants/actionTypes'

const rawdata = (state = [], action) => {
  switch(action.type) {
    case LOAD_DATA:
      return action.data;
    default:
      return state;
  }
};

export default combineReducers({
  rawdata
});
