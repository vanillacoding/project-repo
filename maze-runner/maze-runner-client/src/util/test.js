import { NODE_STATUS } from '../constant';
import { createNodes } from './maze';

export function getNodeColIndex(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length !== 2) {
    throw new Error('Invalid node id');
  }

  return nodeId.split('-')[1];
}

export function getNodeRowIndex(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length !== 2) {
    throw new Error('Invalid node id');
  }

  return nodeId.split('-')[0];
}

export function getPrevNodeId(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length < 2) {
    throw new Error('Invalid node id');
  }

  const row = getNodeRowIndex(nodeId);
  const col = getNodeColIndex(nodeId);

  return `${Number(row)}-${Number(col) - 1}`;
}

export function createById(startNodeId, endNodeId) {
  const exampleBlocks = [
    [0, 0, 0, 1, 0, 0, 1, 0, 0, 1],
    [0, 1, 0, 1, 0, 1, 1, 0, 1, 1],
    [0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
    [1, 1, 0, 1, 1, 1, 1, 1, 1, 1],
    [0, 0, 0, 1, 0, 1, 1, 1, 0, 0],
    [0, 1, 1, 1, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 1, 1, 1, 1, 1, 0],
    [1, 1, 1, 0, 1, 0, 0, 0, 0, 0],
    [1, 0, 0, 0, 1, 0, 1, 1, 0, 1],
    [1, 0, 1, 0, 0, 0, 0, 1, 0, 0],
  ];

  const startNodeIndex = {
    col: getNodeColIndex(startNodeId),
    row: getNodeRowIndex(startNodeId),
  };
  const endNodeIndex = {
    col: getNodeColIndex(endNodeId),
    row: getNodeRowIndex(endNodeId),
  };

  exampleBlocks[startNodeIndex.row][startNodeIndex.col] = 3;
  exampleBlocks[endNodeIndex.row][endNodeIndex.col] = 4;

  const [widthCount, heightCount] = [
    exampleBlocks[0].length,
    exampleBlocks.length,
  ];

  const nodes = createNodes(widthCount, heightCount);
  const { byId } = nodes;

  exampleBlocks.forEach((row, rowIndex) =>
    row.forEach((number, colIndex) => {
      const nodeId = `${rowIndex}-${colIndex}`;

      switch (number) {
        case 0: {
          byId[nodeId].status = NODE_STATUS.UNVISITED;
          return;
        }
        case 1: {
          byId[nodeId].status = NODE_STATUS.WALL;
          return;
        }
        case 2: {
          byId[nodeId].status = NODE_STATUS.WEIGHTED;
          byId[nodeId].weight = 10;
          return;
        }
        case 3: {
          byId[nodeId].status = NODE_STATUS.START;
          return;
        }
        case 4: {
          byId[nodeId].status = NODE_STATUS.END;
          return;
        }
        case 5: {
          byId[nodeId].status = NODE_STATUS.MIDDLE;
          return;
        }
        default: {
          return;
        }
      }
    }),
  );

  return byId;
}

export default {
  getNodeColIndex,
  getNodeRowIndex,
  getPrevNodeId,
  createById,
};
