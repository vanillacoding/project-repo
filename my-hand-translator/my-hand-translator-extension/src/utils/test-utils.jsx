import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import userReducer from "../features/user/userSlice";

function render(
  ui,
  {
    preloadedState,
    store = configureStore({ reducer: { user: userReducer }, preloadedState }),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";

const user = {
  projectId: "translate-324702",
  translations: [
    {
      nanoId: "qWYBIfLPMsjP1LHNRqlFW",
      origin: "react",
      translated: "리액트",
      url: "vanilla",
      glossary: {
        react: "리액트",
      },
      createdAt: "2017-01-01",
    },
  ],
  glossary: {
    react: "리액트",
  },
  glossaryId: "6141bd1c2a11e5b51320bac1",
  isServerOn: false,
  name: "aidencoders",
  email: "aidencoders@gmail.com",
};

export { render, user };
