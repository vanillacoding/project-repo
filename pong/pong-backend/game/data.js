const canvasData = {
  ballData: {
    x: 250,
    y: 250,
    dx: 0,
    dy: 5,
    radius: 10,
    speed: 10,
  },
  userPaddleData: {
    height: 20,
    width: 100,
    x: 250,
  },
  partnerPaddleData: {
    height: 20,
    width: 100,
    x: 250,
  },
};

module.exports = canvasData;
