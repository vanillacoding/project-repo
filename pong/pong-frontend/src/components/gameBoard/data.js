import { COLORS } from "../../constants";

const canvasData = {
  ballObj: {
    x: 250,
    y: 250,
    dx: 0,
    dy: 5,
    radius: 10,
    speed: 10,
  },
  userPaddleObj: {
    height: 20,
    width: 100,
    x: 250,
    color: COLORS.USER_PADDLE
  },
  partnerPaddleObj: {
    height: 20,
    width: 100,
    x: 250,
    color: COLORS.PARTNER_PADDLE
  },
};

export default canvasData;
