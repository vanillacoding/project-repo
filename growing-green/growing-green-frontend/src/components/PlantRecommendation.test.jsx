import React from 'react';
import PlantRecommendation from './PlantRecommendation';
import { render } from '../test/test-utils';
import { mockPlantNameList } from '../test/data';

describe('<PlantRecommendation>', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <PlantRecommendation plantsNames={mockPlantNameList.data} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render recommended plant names', () => {
    const { getByText } = render(
      <PlantRecommendation plantsNames={mockPlantNameList.data} />,
    );
    expect(
      getByText(/사용자들이 가장 많이 키우는 식물 TOP 5/i),
    ).toBeInTheDocument();
    mockPlantNameList.data.forEach((name) => {
      expect(getByText(name)).toBeInTheDocument();
    });
  });
});
