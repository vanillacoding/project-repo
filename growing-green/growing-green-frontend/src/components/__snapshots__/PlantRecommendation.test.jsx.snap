// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlantRecommendation> matches snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL rGCdf"
  >
    <h1
      class="sc-gsDKAQ hZXsZz"
    >
      사용자들이 가장 많이 키우는 식물 TOP 5
    </h1>
    <p>
      장미허브 Vicks plant Plectranthus tomentosa
    </p>
    <p>
      리틀장미 Echeveria prolifica
    </p>
    <p>
      장미 Rose Rosa hybrida
    </p>
  </div>
</div>
`;
